events/api/src/main/java/org/projectnessie/events/api/ContentRemovedEvent.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+/**
+ * Event that is emitted when a content is removed. This event corresponds to a DELETE operation in
+ * a commit, merge or transplant. This event is emitted after the content has been removed.
+ */
+@Value.Immutable
+@JsonTypeName("CONTENT_REMOVED")
+@JsonSerialize
+@JsonDeserialize
+public interface ContentRemovedEvent extends ContentEvent {
```

Comment:
The design doc calls for a `Content type` attribute here... Did I miss it? Would it be meaningful at the `ContentEvent` level?

Reply:
Good catch! That said I think the design doc is wrong... we can't know the content type of a DELETE. If you agree, I will amend the design doc in a separate PR.

Reply:
FYI https://github.com/projectnessie/nessie/pull/6672

Reply:
Yes, #6672 makes sense